python_ta/__init__.py

```diff
@@ -150,101 +260,41 @@ def _check(
     """
     # Configuring logger
     logging.basicConfig(format="[%(levelname)s] %(message)s", level=logging.INFO)
-
-    linter = reset_linter(config=local_config, load_default_config=load_default_config)
-    current_reporter = linter.reporter
-    current_reporter.set_output(output)
-    messages_config_path = linter.config.messages_config_path
-    messages_config_default_path = linter._option_dicts["messages-config-path"]["default"]
-    use_pyta_error_messages = linter.config.use_pyta_error_messages
-    messages_config = load_messages_config(
-        messages_config_path, messages_config_default_path, use_pyta_error_messages
-    )
-
-    global PYLINT_PATCHED
-    if not PYLINT_PATCHED:
-        patch_all(
-            messages_config, linter.config.z3
-        )  # Monkeypatch pylint (override certain methods)
-        PYLINT_PATCHED = True
-
-    # Try to check file, issue error message for invalid files.
+    linter, current_reporter = _setup_linter(local_config, load_default_config, output)
     try:
         # Flag indicating whether at least one file has been checked
         is_any_file_checked = False
-
+        linted_files = set()
         for locations in _get_valid_files_to_check(module_name):
             f_paths = []  # Paths to files for data submission
-            errs = []  # Errors caught in files for data submission
-            config = {}  # Configuration settings for data submission
             for file_py in get_file_paths(locations):
-                allowed_pylint = linter.config.allow_pylint_comments
-                if not _verify_pre_check(file_py, allowed_pylint):
-                    continue  # Check the other files
-                # Load config file in user location. Construct new linter each
-                # time, so config options don't bleed to unintended files.
-                # Reuse the same reporter each time to accumulate the results across different files.
-                linter = reset_linter(
-                    config=local_config,
-                    file_linted=file_py,
-                    load_default_config=load_default_config,
+                linted_files.add(file_py)
+                is_any_file_checked, current_reporter, linter = _check_file(
+                    linter,
```

Comment:
In general, for function calls with many arguments, provide parameter names (e.g. `linter=linter`) to make it clear what each argument refers to.